packages/flutter/lib/src/gestures/converter.dart

```diff
@@ -241,3 +242,86 @@ class PointerEventConverter {
   static double _toLogicalPixels(double physicalPixels, double devicePixelRatio) =>
       physicalPixels == null ? null : physicalPixels / devicePixelRatio;
 }
+
+
+
+/// Serializes [ui.PointerData] to a json object.
+///
+/// Default values are omitted in the json object.
+///
+/// See [deserializePointerData].
+Map<String, dynamic> serializePointerData(ui.PointerData data) {
+      // to perform null check on a variable rather than a getter
+  return <String, dynamic>{
+    if(data.timeStamp != Duration.zero) 'timeStamp': data.timeStamp.inMicroseconds,
+    if(data.change != ui.PointerChange.cancel) 'change': data.change.index,
+    if(data.kind != ui.PointerDeviceKind.touch) 'kind': data.kind.index,
+    if(data.signalKind != null) 'signalKind': data.signalKind.index,
+    if(data.device != 0) 'device': data.device,
+    if(data.pointerIdentifier != 0) 'pointerIdentifier': data.pointerIdentifier,
+    if(data.physicalX != 0.0) 'physicalX': data.physicalX,
+    if(data.physicalY != 0.0) 'physicalY': data.physicalY,
+    if(data.physicalDeltaX != 0.0) 'physicalDeltaX': data.physicalDeltaX,
+    if(data.physicalDeltaY != 0.0) 'physicalDeltaY': data.physicalDeltaY,
+    if(data.buttons != 0) 'buttons': data.buttons,
+    if(data.obscured != false) 'obscured': data.obscured,
+    if(data.synthesized != false) 'synthesized': data.synthesized,
+    if(data.pressure != 0.0) 'pressure': data.pressure,
+    if(data.pressureMin != 0.0) 'pressureMin': data.pressureMin,
+    if(data.pressureMax != 0.0) 'pressureMax': data.pressureMax,
+    if(data.distance != 0.0) 'distance': data.distance,
+    if(data.distanceMax != 0.0) 'distanceMax': data.distanceMax,
+    if(data.size != 0.0) 'size': data.size,
+    if(data.radiusMajor != 0.0) 'radiusMajor': data.radiusMajor,
+    if(data.radiusMinor != 0.0) 'radiusMinor': data.radiusMinor,
+    if(data.radiusMin != 0.0) 'radiusMin': data.radiusMin,
+    if(data.radiusMax != 0.0) 'radiusMax': data.radiusMax,
+    if(data.orientation != 0.0) 'orientation': data.orientation,
+    if(data.tilt != 0.0) 'tilt': data.tilt,
+    if(data.platformData != 0) 'platformData': data.platformData,
+    if(data.scrollDeltaX != 0.0) 'scrollDeltaX': data.scrollDeltaX,
+    if(data.scrollDeltaY != 0.0) 'scrollDeltaY': data.scrollDeltaY,
+  };
+}
+
+/// Deserializes [ui.PointerData] from a json object.
+///
+/// See [serializePointerData].
+ui.PointerData deserializePointerData(dynamic value) {
```

Comment:
dynamic is strange. If this is the opposite of serializePointerData it should just take whatever that method returns.